app/models/solidus_paypal_braintree/source.rb

```diff
@@ -80,11 +82,23 @@ def display_number
 
     private
 
-    def braintree_payment_method
+    def persist_credit_card_info
+      create_payment_details!(
+        last_4: @payment_details.last_4,
+        expiration_month: @payment_details.expiration_month,
+        expiration_year: @payment_details.expiration_year,
+        card_type: @payment_details.card_type,
+        cardholder_name: @payment_details.cardholder_name
```

Comment:
Because this is a polymorphic association, you'll need to specify the `type` as well.
```suggestion
        cardholder_name: @payment_details.cardholder_name,
        type: "SolidusPaypalBraintree::CreditCard"
```

Reply:
Seems like there is no such method. 
For the non polymorphic relations rails is creating a create_ method but not for the polymorphic relation. So i guess i will revert to the previous way of creating record and associating it.
0> pp self.methods.select{|x| x.to_s.include? 'create_payment_details'} 
=> []

Reply:
Oh right. That's because this is a polymorphic association which have a whole bunch of limitations. We'll have to go back to the long form.

Reply:
Already made the changes and added tests.